### DISCUSSION TITLE
Improve code reading experience in Clion

### DISCUSSION THREAD

**Discussion by chaohuzhang** [pocoproject/poco#3747] [General]
When I open it with Clion, all the files are mixed together, and the directory structure is clear when I open it with visual studio

**Comment by aleks-f** [comment]
If possible at all, someone with CMake knowledge will have to step in here.

**Comment by JackyWoo** [comment]
@chaohuzhang I use Clion and it works well. Could you pls describe in details?

**Comment by aleks-f** [comment]
@JackyWoo what he is complaining about is that in CLion files are not grouped like they are in Visual Studio

![image](https://user-images.githubusercontent.com/2429093/184503652-3b5318c3-2481-4afd-92bb-c2dd7bb1438a.png)

**Reply by JackyWoo** [reply]
Got it, thanks.

**Comment by xakod** [comment]
You need to use cmake ```source_group``` https://cmake.org/cmake/help/latest/command/source_group.html for ide source groups
